crawl-ref/source/dat/des/variable/large_abstract.des

```diff
@@ -1750,3 +1750,183 @@ x0xx.xx0x.x.0x%x.xxxx0x....x...xx..xx..xx..xx.....x.x..x...x..xx....+...+.000%%x
 x..0.0..x.x...........xxx.0.0x.0..xx0...0.xx0.....x0..0x.0..0...0.0.x000x000x%%x
 xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 ENDMAP
+
+# The nh_big_room_* vaults are a NetHack reference/homage, so the real credit is
+# to whichever NetHack dev made the Big Room levels. I wanted these vaults to be
+# ORIENT: centre, but they're too wide for that, so the game will just refuse to
+# load the maps at startup. So we'll fake it by making them ORIENT: north and
+# then using spacers to shove them back towards the middle. The big_room vaults
```

Comment:
Dubious... How over the limit are these on width for `ORIENT: centre`? Is it really impossible to change shape? 

Alternatively, what does the level builder complain about them exactly, and can we work around that restriction from the C++ side... (willing to dive into this, but not at my crawlputer now).

Reply:
> Dubious... How over the limit are these on width for ORIENT: centre? Is it really impossible to change shape?

It's not impossible, but just porting over the Nethack vault square-by-square makes them this wide. I suppoooose I could make some editorial changes to make them smaller but keeping the same basic shape...

> Alternatively, what does the level builder complain about them exactly, and can we work around that restriction from the C++ side... (willing to dive into this, but not at my crawlputer now).

It's [this bit in mapdef.cc](https://github.com/crawl/crawl/blob/5b5c5c9dfb8e383e54acd0718f9f1718c98ca31b/crawl-ref/source/mapdef.cc#L2934); for vaults oriented in the four cardinal directions, size-checking is only checked along one dimension. I'm generally reluctant to make too many changes to the code base just to support some goofy vault idea, just out of a sense of not wanting to overstep my bounds (so to speak), but if you think it's worthwhile I guess we can do that.

For what it's worth, I didn't notice any major issues during testing with the hack-around used here. (Although since I wasn't in debug mode, I guess if a bunch of levels got vetoed before the final one appeared, I wouldn't have noticed.)

Reply:
I updated based on the other comments, but I'm still not sure what to do with this one.

Reply:
Hmmmm. I'll ping @rawlins about this, since he's done a lot of work on reducing vetoes. Reading the builder log this seems to work, I'm out of time now but I'll try to git blame the maximum dimensions and see if I can find some reasons for them to be set what they are.